File: java-api/src/main/java/com/fagerland/javaapi/models/TimeEntry.java
package com.fagerland.javaapi.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
@Table(name = "timeEntry")
public class TimeEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private long startStamp;
    private long endStamp;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "userEntry_id", nullable = false)
    private UserEntry userEntry;

    public TimeEntry(long startStamp, long endStamp, UserEntry userEntry) {
        this.startStamp = startStamp;
        this.endStamp = endStamp;
        this.userEntry = userEntry;
    }

    public TimeEntry() {
    }

    public Long getId() {
        return id;
    }

    public UserEntry getUserEntry() {
        return userEntry;
    }

    public void setUserEntry(UserEntry userEntry) {
        this.userEntry = userEntry;
    }

    public long getStartStamp() {
        return startStamp;
    }

    public long getEndStamp() {
        return endStamp;
    }

    public void setEndStamp(long endStamp) {
        this.endStamp = endStamp;
    }
}

File: kotlin-maven-api/src/main/kotlin/com/fagerland/brotime/models/TimeEntry.kt
package com.fagerland.brotime.models

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "timeEntry")
class TimeEntry(
        var startTime: Long?,
        var endTime: Long?,
        var timeZone: String?,

        @JsonIgnore
        @ManyToOne
        var userEntry: UserEntry,

        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long? = null
)